modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/CoordinationConfigurableClusterAgent.java

```diff
@@ -204,16 +204,52 @@ public void start(ClusterManager manager) throws AndesException{
             log.debug("Unique ID generation for message ID generation:" + uniqueIdOfLocalMember);
         }
 
-        String thriftCoordinatorServerIP = AndesContext.getInstance().getThriftServerHost();
-        int thriftCoordinatorServerPort = AndesContext.getInstance().getThriftServerPort();
-        InetSocketAddress thriftAddress = new InetSocketAddress(thriftCoordinatorServerIP, thriftCoordinatorServerPort);
+        InetSocketAddress thriftAddress = new InetSocketAddress(getThriftCoordinatorServerAddress(), getThriftCoordinatorServerPort());
 
         coordinationStrategy.start(this, getLocalNodeIdentifier(), thriftAddress);
 
         networkPartitionDetector.start();
     }
 
     /**
+     * Get the routable IP address of Thrift Coordinator Server
+     *
+     * @return  IP address for Thrift Coordinator Server
+     */
+    private String getThriftCoordinatorServerAddress() {
+        // first check if the advertised routable ip is present
+        String thriftCoordinatorServerIP = AndesContext.getInstance().getThriftServerAdvertisedHost();
+        if (!thriftCoordinatorServerIP.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Thrift Coordinator Server advertised IP specified as " + thriftCoordinatorServerIP);
```

Comment:
I think if we can move this debug log statement to the line before returning the thriftCoordinatorServerIP, it would print the server ip in both cases. Thanks!